### PR TITLE
fixing bugs and removing leptons with deltaR > 0.4

### PR DESCRIPTION
**Pull request by hayfasfar**

### CODE REVIEW

File: NANOProducer/plugins/NANOProducer.cc [hunk 1]
```diff
@@ -212,7 +212,7 @@ NANOProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
     std::vector<float> mu_dzError;
     std::vector<float> mu_numberOfValidPixelHits;
     std::vector<float> mu_numberOfpixelLayersWithMeasurement;
-    //std::vector<float> mu_numberOfstripLayersWithMeasurement ;
+    std::vector<float> mu_numberOfstripLayersWithMeasurement ;
```
Comment (matt-komm) [inline NANOProducer/plugins/NANOProducer.cc]: are these floats or ints?

File: XTagInfoProducer/plugins/XTagInfoProducer.cc [hunk 2]
```diff
@@ -472,24 +471,31 @@ XTagInfoProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
...
+                elec_features.elec_jetDeltaR = reco::deltaR(electron,jet); 
                 elec_features.elec_EtFromCaloEn = electron.caloEnergy() * sin(electron.p4().theta());
                 elec_features.elec_ecalDrivenSeed = electron.ecalDrivenSeed();
 
                 elec_features.elec_isEB = electron.isEB();
                 elec_features.elec_isEE  = electron.isEE();
                 elec_features.elec_ecalEnergy  = electron.ecalEnergy()/electron.pt();
                 elec_features.elec_isPassConversionVeto = electron.passConversionVeto();
-
+		if(electron.convDist() != -9999){
```
Comment (matt-komm) [inline XTagInfoProducer/plugins/XTagInfoProducer.cc]: note that convDist() returns float so the comparison to int here is dangerous; better 
`if (std::fabs(electron.convDist()+9999)<1e-4)) `
if you want to test for -9999 only; if any negative value make no sense just use
`if (electron.convDist()<0) `
instead

File: XTagInfoProducer/plugins/XTagInfoProducer.cc [hunk 3]
```diff
@@ -558,22 +560,37 @@ XTagInfoProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
...
+                elec_features.elec_numberOfBrems  = electron.numberOfBrems();
+		if(electron.pt() >= 5. ){
+                elec_features.elec_fbrem = electron.fbrem();
                 elec_features.elec_scSigmaEtaEta = electron.scSigmaEtaEta();
                 elec_features.elec_scSigmaIEtaIEta = electron.scSigmaIEtaIEta();
                 elec_features.elec_superClusterFbrem = electron.superClusterFbrem();
-
-                elec_features.elec_scE5x5 = electron.scE5x5 ();
+		}
+		if (electron.pt() < 5.){
```
Comment (matt-komm) [inline XTagInfoProducer/plugins/XTagInfoProducer.cc]: use else here

File: NANOProducer/plugins/NANOProducer.cc (hunk 1 above)
Comment (hayfasfar) [reply NANOProducer/plugins/NANOProducer.cc]: it is int but i have tested it with float and there was no problem.   I thought that we need float variables type for the unpacker.

File: XTagInfoProducer/plugins/XTagInfoProducer.cc (hunk 3 above)
Comment (hayfasfar) [reply XTagInfoProducer/plugins/XTagInfoProducer.cc]: done.

File: DataFormats/interface/ElectronCandidateFeatures.h [hunk 4]
```diff
@@ -65,6 +67,9 @@ struct ElectronCandidateFeatures
     float elec_dz;
     int elec_nbOfMissingHits;
     float elec_gsfCharge;
+    float elec_ndof ;
```
Comment (matt-komm) [inline DataFormats/interface/ElectronCandidateFeatures.h]: may need to be an int instead

**Pull request merged**